__tests__/run.test.ts

```diff
@@ -1,7 +1,199 @@
+import * as io from '@actions/io';
+import * as toolCache from '@actions/tool-cache';
+import * as core from '@actions/core';
+import * as fs from 'fs';
+import * as jsyaml from 'js-yaml';
+import * as path from 'path';
 
-describe('This is a placeholder for intial test cases, to be removed', () => {
-    test('Dummy test case', async () => {
-        await expect(run()).rejects.toThrow();
-    })
-})
\ No newline at end of file
+var mockStatusCode;
+const mockExecFn = jest.fn().mockImplementation(() => mockStatusCode);
+jest.mock('@actions/exec/lib/toolrunner', () => {
+    return {
+        ToolRunner: jest.fn().mockImplementation(() => {
+            return {
+                exec: mockExecFn  
+            }
+        })
+    }
+});
+
+
+describe('Testing all functions.', () => {
+    test('getExecutableExtension() - return .exe when os is Windows', () => {
+        jest.spyOn(os, 'type').mockReturnValue('Windows_NT');
+    
+        expect(run.getExecutableExtension()).toBe('.exe');
+        expect(os.type).toBeCalled();         
+    });
+
+    test('getExecutableExtension() - return empty string for non-windows OS', () => {
+        jest.spyOn(os, 'type').mockReturnValue('Darwin');
+    
+        expect(run.getExecutableExtension()).toBe('');         
+        expect(os.type).toBeCalled();         
+    });
+
+    test('getKubectlPath() - return path to existing kubectl', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('pathToKubectl');
+
+        expect(await run.getKubectlPath()).toBe('pathToKubectl');
+        expect(io.which).toBeCalled();         
+    });
+
+    test('getKubectlPath() - throw error when kubectl not installed', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('');
+        jest.spyOn(toolCache, 'findAllVersions').mockReturnValue([]);
+
+        await expect(run.getKubectlPath()).rejects.toThrow('Kubectl is not installed');
+        expect(io.which).toBeCalled();         
+        expect(toolCache.findAllVersions).toBeCalled();         
+    });
+
+    test('getKubectlPath() - return path to kubectl in toolCache', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('');
+        jest.spyOn(toolCache, 'findAllVersions').mockReturnValue(['v1.15.0']);
+        jest.spyOn(toolCache, 'find').mockReturnValue('pathToTool');
+        jest.spyOn(os, 'type').mockReturnValue('Windows_NT');
+
+        expect(await run.getKubectlPath()).toBe(path.join('pathToTool', 'kubectl.exe'));
+        expect(io.which).toBeCalled();         
+        expect(toolCache.findAllVersions).toBeCalled();         
+        expect(toolCache.find).toBeCalled();         
+    });
+    
+    test('getKubeconfig() - throw error on invalid input', () => {
+        jest.spyOn(core, 'getInput').mockReturnValue('invalid');
+
+        expect(() => run.getKubeconfig()).toThrow('Invalid method specified. Acceptable values are kubeconfig and service-account.');
+        expect(core.getInput).toBeCalled();         
```

Comment:
I think this test is causing an error in the [check](https://github.com/Azure/k8s-set-context/runs/2112892832?check_suite_focus=true)

Reply:
Im not sure but i dont think its just one of the tests. commenting out [this](https://github.com/sundargs2000/k8s-set-context/blob/e55248ff72de1edf6a8933343f934d66370ca3d2/src/login.ts#L116) line seems to stop the error being shown. even [these](https://github.com/Azure/k8s-deploy/actions/runs/664527942) tests have them. not sure how we can avoid them.

Reply:
I think the reason is `getInput` function has `method` as a required parameter. Refer the [getInput](https://github.com/actions/toolkit/blob/4dd900dde00dcc90e5c649b688c643b1fc7684a2/packages/core/src/core.ts#L84) method

If we are testing an invalid value for `method` here, we still need to pass method as a parameter.

```typescript
jest.spyOn(core, 'getInput').mockImplementation((inputName, options) => {
     if (inputName == 'method') return 'invalidMethod';
});

expect(() => run.getKubeconfig()).toThrow('Invalid method specified. Acceptable values are kubeconfig and service-account.');
expect(core.getInput).toBeCalledWith('method', { required: true }); 
 ```

Reply:
It is still showing that error 🙁